### ISSUE TITLE
Whitespace padding in object literals, destructuring assignment, and import statements

### ISSUE THREAD

**Issue by leeavital** [palantir/tslint#1044]
Enforce whitespace padding in object literals:

``` ts
{ myField } // good
{ myField: "hello" } // good
{myField} // bad
{myField: "hello"} //bad
```

**Comment by vsiao** [comment]
I assume the same style should apply to named imports.

**Comment by Diokuz** [comment]
Any progress?

**Comment by filipsuk** [comment]
+1

**Comment by xuhong** [comment]
+1

**Comment by giladgray** [comment]
an option for the same thing for arrays would be great too. `[a, b]` vs `[ a, b ]`

**Comment by glen-84** [comment]
I prefer it without the spaces, so an option for that would also be nice.

**Comment by danieljin** [comment]
+1

**Comment by blingerson** [comment]
It would also be nice if we could enforce spacing before and after the colon like

`
{ foo : 'bar' } //good
{ foo: 'bar' } //bad
{ foo:'bar' } //bad
`

**Comment by irek02** [comment]
Any progress on this? I would really like to use this in our projects at Acquia.

**Comment by js2me** [comment]
OMG, it still open?

**Comment by pelotom** [comment]
![photofunia-1532138160](https://user-images.githubusercontent.com/128019/43031042-c6d99686-8c67-11e8-9324-1a338fe1628d.jpg)

**Comment by peterjwest** [comment]
BTW tslint-eslint-rules supports in nearly all cases (not in object literal types) with the rule `object-curly-spacing`. Perhaps this could be ported to tslint?

**Comment by MarcusNoble** [comment]
we use the following to achieve this:
```json
{
    "extends": [
        "tslint-eslint-rules"
    ],
    "rules": {
        "object-curly-spacing": true,
        "array-bracket-spacing": [
            true,
            "never"
        ]
    }
}
```

**Comment by micheleangioni** [comment]
Any update on this? :)

**Comment by js2me** [comment]
@micheleangioni I'm just using eslint plugin for integration tslint with eslint and that solves my problem linked with this issue

**Comment by toupeira** [comment]
I tried out `object-curly-spacing` from `tslint-eslint-rules` but it doesn't seem to cover empty objects (`{}` vs. `{ }`), unless I misconfigured something.

Would be great to have something like this in tslint proper!

**Comment by toupeira** [comment]
I see https://github.com/palantir/tslint/pull/4068 added a check for spaces after open braces, but it doesn't detect more than one space, and doesn't deal with closing braces and empty objects.

**Comment by adidahiya** [comment]
There have been very few PRs submitted to address this issue, and it is not a priority for TSLint core (see #4534), so I'm going to close this. If you want to implement this whitespace check, it can happily live in a custom rule set outside this repo.

**Comment by puneet-sutar** [comment]
I was looking for the solution and a colleague suggested it is already part of TSlint.
https://palantir.github.io/tslint/rules/whitespace/

**Comment by joshi-anshul** [comment]
You just need to add below rules in tslint.json file:

`    "whitespace": {
      "options": [
        "check-branch",
        "check-decl",
        "check-operator",
        "check-separator",
        "check-type",
        "check-typecast",
        "check-module",
        "check-rest-spread",
        "check-preblock",
        "check-postbrace"
      ]
    },`
Here `        "check-preblock",
        "check-postbrace"` this rules enforce to add white space after braces

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
